engine/robot.py

```diff
@@ -253,22 +262,25 @@ def execute_setup(self):
             self.robot_state.ekf = LocalizationEKF(mu, sigma)
 
             if (self.robot_state.radio_session.connected and gps_setup and imu_setup):
-                obstacle_avoidance = threading.Thread(target=self.track_obstacle, daemon=True)
+                obstacle_avoidance = threading.Thread(
+                    target=self.track_obstacle, daemon=True)
                 obstacle_avoidance.start()  # spawn thread to monitor obstacles
                 self.set_phase(Phase.TRAVERSE)
         else:
             self.set_phase(Phase.TRAVERSE)
 
     def execute_traversal(self, unvisited_waypoints, allowed_dist_error, base_station_loc, control_mode, time_limit,
                           roomba_radius, database):
+        self.robot_state.prev_phase = Phase.TRAVERSE
         self.robot_state.control_mode = control_mode
         if control_mode == 4:  # Roomba mode
+            # roomba mode uses its own obstacle avoidance
+            self.robot_state.enable_obstacle_avoidance = False
```

Comment:
is there a reason we set enable_obstacle_avoidance to False then to True? It doesn't look like this attribute is used in traverse_roomba.

Reply:
this attribute is used in track_obstacle. We want to "pause" the obstacle avoidance thread bc roomba mode has its own obstacle avoidance